### PR TITLE
Improve the `close` method in the graph

### PR DESCRIPTION
**Pull request by SimFG**
issue: #19099
Signed-off-by: SimFG <bang.fu@zilliz.com>

### CODE REVIEW

**Comment by SimFG** [comment]
rerun ut

**Comment by SimFG** [comment]
rerun ut

**Comment by SimFG** [comment]
rerun ut

**Comment by SimFG** [comment]
rerun ut

File: internal/util/flowgraph/node.go [hunk 1]
```diff
@@ -98,8 +103,17 @@ func (nodeCtx *nodeCtx) work() {
 				nodeCtx.collectInputMessages()
 				inputs = nodeCtx.inputMessages
 			}
-			n := nodeCtx.node
-			res = n.Operate(inputs)
+			if isCloseMsg(inputs) {
```
Comment (bigsheeper) [inline internal/util/flowgraph/node.go:106]: a little bit confused here, how about:
```
if isCloseMsg(inputs) {
    res = inputs
    close(nodeCtx.closeCh)
    nodeCtx.node.Close()
} else {
    n := nodeCtx.node
    res = n.Operate(inputs)
}
```

Comment (SimFG) [reply internal/util/flowgraph/node.go:106]: you can understand the input node decides whether the `operate` method is executed and the `res` decide whether the node should be closed.

**Comment by SimFG** [comment]
rerun ut

**Pull request merged**